### PR TITLE
Fix incorrect updating of user's role when OIDC role mapping is not set

### LINKED ISSUE #24247: OpenID Auth: user role reverts to default when signing in with SSO
### Describe the Bug

Since we upgraded to Directus v11.3.4, all existing users that sign in using SSO are assigned the default role set in the `AUTH_<provider>_DEFAULT_ROLE_ID`. That means even admins are being degraded to the default user role!

I've done some research, it looks like this commit from last week is causing this: https://github.com/directus/directus/commit/feeda04ad8046b27b6965cefb557db2fa35c786e#diff-160e58a4a442b2dfcc47f5e9b61aae24dd3a9c83be5ec10b464a9d6ca00f524f

In particular, lines 200 (getting default role from config) and line 248 (setting role in emitPayload that's later used to update the user with) seem to be the ones to blame.

### To Reproduce

Sign in with OpenID SSO as a user with a different role than specified in the `AUTH_<provider>_DEFAULT_ROLE_ID` environment variable. The user's role will be reset to the default.

### Directus Version

v11.3.4

### Hosting Strategy

Self-Hosted (Docker Image)

### Database

_No response_

### PR DESCRIPTION
**Pull request by licitdev**
## Scope

What's changed:

- Update user's role only when OIDC role mapping is set

## Potential Risks / Drawbacks

- Lorem ipsum dolor sit amet
- Consectetur adipiscing elit

## Review Notes / Questions

- I would like to lorem ipsum
- Special attention should be paid to dolor sit amet

---

Fixes #24247

### CODE REVIEW

**Review by rijkvanzanten — APPROVED** [review]

File: api/src/auth/drivers/openid.ts [hunk 1]
```diff
@@ -244,10 +244,13 @@ export class OpenIDAuthDriver extends LocalAuthDriver {
 			// user that is about to be updated
 			let emitPayload: Record<string, unknown> = {
 				auth_data: userPayload.auth_data,
-				// Make sure a user's role gets updated if his openid group or role mapping changes
-				role: role,
 			};
 
+			// Make sure a user's role gets updated if his openid group or role mapping changes
```
Comment (paescuj) [inline api/src/auth/drivers/openid.ts:249]:
```suggestion
			// Make sure a user's role gets updated if their openid group or role mapping changes
```

File: .changeset/wet-laws-sort.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@directus/api': patch
+---
+
+Updated user's role only when OIDC role mapping is set
```
Comment (ComfortablyCoding) [inline .changeset/wet-laws-sort.md:5]:
```suggestion
Fixed users role updating when OIDC role mapping was not present
```

**Review by licitdev — COMMENTED** [review]
I have tested the role to not be replaced when role mapping isn't set.

**Review by ComfortablyCoding — APPROVED** [review]
LGtM 🚀

**Pull request merged**